lasy/utils/laser_utils.py

```diff
@@ -1009,8 +1009,11 @@ def get_zeta(dim, grid, k0):
     # Calculate dx0 and dy0 in (x,y,omega) space
     weight_x_3d = np.transpose(env_spec_abs2, (2, 1, 0))
     weight_y_3d = np.transpose(env_spec_abs2, (2, 0, 1))
-    xda = np.sum(grid.axes[0] * weight_x_3d, axis=2) / np.sum(weight_x_3d, axis=2)
-    yda = np.sum(grid.axes[1] * weight_y_3d, axis=2) / np.sum(weight_y_3d, axis=2)
+    weight_x_2d = np.sum(weight_x_3d, axis=2)
+    weight_y_2d = np.sum(weight_y_3d, axis=2)
+    eps_wgt = 1e-20 if np.any(weight_x_2d == 0) or np.any(weight_y_2d == 0) else 0
+    xda = np.sum(grid.axes[0] * weight_x_3d, axis=2) / (weight_x_2d + eps_wgt)
+    yda = np.sum(grid.axes[1] * weight_y_3d, axis=2) / (weight_y_2d + eps_wgt)
```

Comment:
Not tested, but would sth like this work? Arbitrary values should be avoided when possible.
```suggestion
    xda = np.where(weight_x_2d == 0, 0, np.sum(grid.axes[0] * weight_x_3d, axis=2) / weight_x_2d)
    yda = np.where(weight_y_2d == 0, 0, np.sum(grid.axes[1] * weight_y_3d, axis=2) / weight_y_2d)
```

Reply:
Thanks for the point! this works!